Sources/WalletConnectKMS/Serialiser/Envelope.swift

```diff
@@ -0,0 +1,74 @@
+import Foundation
+import WalletConnectUtils
+
+/// A type representing envelope with it's serialization policy
+public struct Envelope: Equatable {
+    enum Errors: String, Error {
+        case malformedEnvelope
+        case unsupportedEnvelopeType
+    }
+
+    let type: EnvelopeType
+    let sealbox: Data
+
+    /// - Parameter base64encoded: base64encoded envelope
+    /// tp = type byte (1 byte)
+    /// pk = public key (32 bytes)
+    /// iv = initialization vector (12 bytes)
+    /// ct = ciphertext (N bytes)
+    /// sealbox = iv + ct + tag
+    /// type0: tp + sealbox
+    /// type1: tp + pk + sealbox
+    init(_ base64encoded: String) throws {
+        guard let envelopeData = Data(base64Encoded: base64encoded) else {
+            throw Errors.malformedEnvelope
+        }
+        let envelopeTypeByte = envelopeData.subdata(in: 0..<1).uint8
+        if envelopeTypeByte == 0 {
+            self.type = .type0
+            self.sealbox = envelopeData.subdata(in: 1..<envelopeData.count)
+        } else if envelopeTypeByte == 1 {
+            let pubKey = envelopeData.subdata(in: 1..<33)
```

Comment:
does it crash if data length is less than 33?

Reply:
you're correct